htdocs/install/inc.php

```diff
@@ -70,6 +71,67 @@
 //$conffile = "/etc/dolibarr/conf.php";
 //$conffiletoshow = "/etc/dolibarr/conf.php";
 
+$short_options = "c:h";
+$long_options = array(
+	"config:",
+	"help",
+);
+
+/**
+ * Print the usage when executing scripts from install/.
+ *
+ * Print the help text exposing the available options when executing
+ * update or install script (ie. from htdocs/install/) from CLI with
+ * the `php` executable. This function does not `exit` the program and
+ * the caller should then call `exit` themselves since they should
+ * determine whether it was an error or not.
+ *
+ * @param string $program the script that was originally run
+ * @param string $header  the message to signal to the user
+ * @return void
+ */
+function usage($program, $header)
+{
+	echo $header."\n";
+	echo "  php ".$program." [options] previous_version new_version\n";
+	echo "\n";
+	echo "Options:\n";
+	echo "  -c, --config <filename>:\n";
+	echo "      Provide a different conf.php file to use.\n";
+	echo "\n";
+	echo "  -h, --help:\n";
+	echo "      Display this help message.\n";
+}
+
+if (php_sapi_name() === "cli") {
+	$rest_index = 0;
+	$opts = getopt($short_options, $long_options, $rest_index);
+	if ($rest_index < $argc - 2) {
```

Comment:
There is another case that can be used when upgrade2 is ran as a CLI programm. this is the syntax:

upgrade2.php  0.0.0  0.0.0  MAIN_MODULE_XXX,MAIN_MODULE_YYY,...
So 3 paramaters and $argv[3] is used in such a case.
Can you fix your test to allow this case ?

Reply:
Sure, I'll fix that and check.

In the mean time, do think it's ok to keep the cli parsing there? What I think about was dedicating a cli.php which would be including upgrade.php and upgrade.php2 depending on what's needed, allowing cli.php to be in scripts/ instead, but that's currently a bit convoluted because of how inc.php is provided for the upgrade scripts. I'm not sure where I should discuss this, maybe next DevCamp?

Reply:
CLI parsing inside the inc.php is a fist good step.

Moving it into a  script/dolibarr-tools.php may be possible. But it should be a wrapper that execute the most as possible existing codes and not contains logical business code, just a launcher to call others.

Reply:
Ok, done with a few fixes (short options) and changes (moving options down to allow parsing -h directly). Do I need to add more tests for this in travis?